crates/mfem/src/lib.rs

```diff
@@ -183,54 +141,79 @@ pub enum RefAlgo {
     B = 1,
 }
 
-pub use mfem_sys::ffi::BasisType;
+pub use mfem_sys::BasisType;
 
 /////////////////////////////
 // FiniteElementCollection //
 /////////////////////////////
 
-pub trait FiniteElementCollection: AsBase<mfem_sys::ffi::FiniteElementCollection> {
+pub struct FiniteElementCollection {
+    inner: UniquePtr<mfem_sys::FiniteElementCollection>,
+}
+
+pub struct FiniteElementCollectionRef<'a> {
+    inner: &'a mfem_sys::FiniteElementCollection,
+}
+
+impl Deref for FiniteElementCollection {
+    type Target = FiniteElementCollectionRef<'a>;
```

Comment:
I tried this approach but it clearly doesn't work.

When declaring `Target =` for the `Deref` instance below, the `'a` lifetime is needed, but there's nowhere for it to come from...

If I understand your code correctly, you work around this by wrapping a pointer instead, so that the lifetime parameter does not appear.

Is that correct?

Reply:
Yes.  The Deref Target cannot have lifetimes that are not already present in the initial structure.  

I have to finish other things first but I'm working on a proposal with a generic `Owned<T>` (similar to `UniquePtr<T>`) but that can carry wrappers around mfem pointers (which we need to define our own methods).  This should avoid the need to have two separate types: there would be `FiniteElementCollection` (holding transparently the mfem-sys value, to be manipulated only be references) and `Owned<FiniteElementCollection>`.  `Owned<T>` would deref to `T` (`ToOwned` can probably implemented for several `T` which possess the adequate C++ constructor) and all methods would be implemented on `T`.  What do you think?

Reply:
That sounds workable! I'll try it here to see how far I get! :)